### DISCUSSION TITLE
Google Sign-In: I can't find "client-secret-file.json"

### DISCUSSION THREAD

**Discussion by sharktankful** [AsifArmanRahman/firebase-rest-api#22] [Q&A] [answered]
To setup google sign in authentification with the module, the documentation is asking me to use "client-secret-file.json" as follows:
```
# Get a reference to the auth service with provider secret file
auth = firebaseApp.auth(client_secret='client-secret-file.json')

# Reference to auth service with provider secret from env variable
client_secret_config = {
   "client_id": environ.get("CLIENT_ID"),
   "client_secret": environ.get("CLIENT_SECRET"),
   "redirect_uris": [environ.get("REDIRECT_URI")]
}

auth = firebaseApp.auth(client_secret=client_secret_config)
```

The problem is that I don't know where in firebase to find "client-secret-file.json". I'm looking all over firebase to see if its there in one of the settings to copy but I can't find it.

**Comment by AsifArmanRahman** [comment, marked as answer]
You can follow instructions provided by Google from [here](https://developers.google.com/identity/protocols/oauth2/web-server#creatingcred). Also, the first time I created the credentials myself, I saw a video from YouTube, link of which I'm sharing [here](https://youtu.be/PKLG5pfs4nY?si=vtpuAERV6WILz0ak) as well.
